File: claude-experiments/dynamic-toolkit/crates/wasm2dynir/as-programs/primes.ts
// Count primes up to n using trial division
export function count_primes(n: i32): i32 {
  let count: i32 = 0;
  for (let i: i32 = 2; i <= n; i++) {
    if (is_prime(i)) {
      count++;
    }
  }
  return count;
}

function is_prime(n: i32): bool {
  if (n < 2) return false;
  for (let i: i32 = 2; i * i <= n; i++) {
    if (n % i == 0) return false;
  }
  return true;
}

File: claude-experiments/oo-lang/bench/primes.py
import time
def is_prime(n):
    if n < 2: return False
    d = 2
    while d*d <= n:
        if n % d == 0: return False
        d += 1
    return True
t0 = time.time()
count = sum(1 for n in range(2, 200000) if is_prime(n))
t1 = time.time()
print(f"primes<200000 = {count}  [{(t1-t0)*1000:.1f} ms]")
